packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -361,11 +361,16 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           // its last updated timestamp to force the browser to fetch the most
           // up-to-date version of this module.
           try {
-            // delay setting `isSelfAccepting` until the file is actually used (#7870)
             const depModule = await moduleGraph.ensureEntryFromUrl(
               unwrapId(url),
               ssr,
-              canSkipImportAnalysis(url) || forceSkipImportAnalysis
+              // delay setting `isSelfAccepting` until the file is actually used (#7870)
+              {
+                isSelfAccepting:
+                  canSkipImportAnalysis(url) || forceSkipImportAnalysis
+                    ? false
+                    : undefined
+              }
```

Comment:
I'd prefer to avoid an extra object allocation for now.

Btw, I handled this change in #7477. Maybe we can close this PR in favor of that, since it's not an urgent change?

Reply:
I don't think the extra allocation has any perf implications given everything else that is run when we create a new ModuleNode.
But seeing the errors in CI, the ModuleNode is a public API, and the `setIsSelfAccepted` param is documented at least in the code. So this is a small breaking change. I think forcing everybody to set `false` when creating a node is too disruptive. I'll close this PR, but I think we should leave `setIsSelfAccepted` untouched in your PR too. I don't think it is worth the disruption, I thought before that it would be something internal to core only.

Reply:
It may be a public API, but it's also undocumented. If anything, we should add `private` to the constructor and see if anyone complains in the next alpha. 🤞

Reply:
Ah I just realized that `setIsSelfAccepted` isn't only a constructor argument. You might be right that it's just too dangerous to change the default.